src/raftstore/store/snap.rs

```diff
@@ -720,27 +1696,50 @@ impl SnapManagerCore {
         self.registry.contains_key(key)
     }
 
-    pub fn get_snapshot_to_build(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
-        let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), true, key));
-        Ok(Box::new(f))
+    pub fn get_snapshot_for_building(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
+        if self.use_sst_file_snapshot {
+            let f = try!(v2::Snap::new_for_building(&self.base, key, self.snap_size.clone()));
+            Ok(Box::new(f))
+        } else {
+            let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), true, key));
+            Ok(Box::new(f))
+        }
     }
 
-    pub fn get_snapshot_for_sending(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
-        let reader = try!(v1::Snap::new_for_reading(&self.base, self.snap_size.clone(), true, key));
-        Ok(Box::new(reader))
+    pub fn get_snapshot_for_sending(&self, key: &SnapKey) -> RaftStoreResult<Box<Snapshot>> {
+        let s = try!(v1::Snap::new_for_reading(&self.base, self.snap_size.clone(), true, key));
```

Comment:
It's for version compatibility. Assume there is a peer which has just generated a v1 snapshot right before it's upgraded. After the upgrade, this v1 snapshot should be used for replication.